Automattic/sensei
# Add course outline basic setup

### Changes proposed in this Pull Request

* Add the course outline basic setup.

It doesn't handle the saving (it has the first part of a way to do it though), the styles, and the data fetches.

About unit tests, I didn't see any very testable part of this code. But if someone has a different opinion, feel free to comment and we can test some parts if it's considered important (test if the files were enqueued, the category was created, if the blocks were registered???).

### Testing instructions

* Go to the course page and make sure the "Outline" block doesn't appear yet (it's behind a feature flag).
* Add the feature flag to your code: `define( 'SENSEI_FEATURE_FLAG_COURSE_OUTLINE', true );`.
* Create a new course. It should start with the Outline block.
* Remove / Add the Outline block and make sure it's restricted to only one per post.
* Add to the "Outline" block "Module" blocks and "Lesson" blocks.
